crates/polars-io/src/csv/write_impl.rs

```diff
@@ -24,9 +24,10 @@ use super::write::QuoteStyle;
 fn fmt_and_escape_str(f: &mut Vec<u8>, v: &str, options: &SerializeOptions) -> std::io::Result<()> {
     if v.is_empty() {
         write!(f, "\"\"")
+    } else if options.quote_style == QuoteStyle::Never {
+        write!(f, "{v}")
```

Comment:
I don't think this clause is necessary, and while it may speed up writing when QuoteStyle is 'never', it will slow down other options. I don't think the 'never' option is popular enough to warrant a 'fast path' like this.

Reply:
Very happy with @stinodego's skepticism regards extra branches. :)

I think in this case we do need a branch somewhere as we also shouldn't follow the `needs_escaping` logic. So having this branch up-front hopefully makes it easy for the branch prediction as it is a constant in the loop.